### ISSUE TITLE
custom login meteor accounts like (saml/oauth)

### ISSUE THREAD

**Issue by Gobliins** [wekan/wekan#708]
Hello, 

i want to add a saml login next to the existing login by including this https://github.com/steffow/meteor-accounts-saml lib.
In my old project i put the {{samllogin}} analog to {{loginbuttons}} on my main html page. However i am totally confused where to include the login at all.

Can someone help me here?

**Comment by xet7** [comment]
This issue is for SAML on Docker/source install.

On Sandstorm platform, SAML is already available, screenshots at:
https://discourse.wekan.io/t/sso-passing-variables-through-url/493/4

**Comment by gerbsen** [comment]
hey there, I don‘t really understand the wiki entry and this issue here, so I better ask again, could you guys tell me the current status of the SAML integration? It’s available for sandstorm but not in the open source version? Is there a way to bring/buy this to be open source? Am I missing something? Thanks, Daniel

**Comment by xet7** [comment]
@gerbsen 

All of Sandstorm is Open Source, including SAML integration. I have tested it, and it works.
https://sandstorm.io/news/2017-02-06-sandstorm-returning-to-community-roots

**Comment by xet7** [comment]
@gerbsen 

Currently only way to support Sandstorm financially is to upgrade to paid account at https://sandstorm.io cloud service, it's highly recommended even if you don't use it, just to keep it running, as Sandstorm developers keep updating it at their free time on weekends.

Supporting Wekan is detailed at https://wekan.team

**Comment by xet7** [comment]
@danpatdav

Yes, adding some SAML package to Wekan would make SAML login possible:

https://atmospherejs.com/?q=saml

https://guide.meteor.com/accounts.html

**Comment by peterk** [comment]
SAML support would unlock Wekan for use in a lot of organizations. We checked the Sandstorm version but it does not work for end users (they find the extra menus confusing). Is anyone already working on adding SAML in Wekan?

**Comment by xet7** [comment]
I will start implementing SAML at about 2020-02-02, it's [at my Roadmap](https://boards.wekan.team/b/D2SzJKZDS4Z48yeQH/wekan-open-source-kanban-board-with-mit-license) Commercial Support list.

**Comment by Vinc89** [comment]
Hello,
just want to give some feedback to the SAML feature.
I set the environment variables in the docker-compose file. Using the quay.io/wekan/wekan:latest image.
SAML_ENABLED=true
SAML_PROVIDER=openam
SAML_ENTRYPOINT=https://ourdomain.com
SAML_ISSUER=https://ourdomain.com
SAML_CERT=MIIG9DCCBd.*
SAML_IDPSLO_REDIRECTURL=https://ourdomain.com
SAML_PRIVATE_KEYFILE=/build/saml-credendials/sp-cert.pem
SAML_PUBLIC_CERTFILE=/build/saml-credendials/sp-key.pem
SAML_IDENTIFIER_FORMAT=urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified
SAML_LOCAL_PROFILE_MATCH_ATTRIBUTE=mail
SAML_ATTRIBUTES=[uid, sn, givenName, mail]

No SAML login button is available. Am I maybe missing some other settings? Thanks in advance!

**Comment by xet7** [comment]
Are browsers going to break SAML and other login systems?
https://news.ycombinator.com/item?id=26728596

I'm just thinking, what login system is worth implementing at all, is it's just going to break.

**Comment by mistial-dev** [comment]
> Are browsers going to break SAML and other login systems? https://news.ycombinator.com/item?id=26728596
> 
> I'm just thinking, what login system is worth implementing at all, is it just going to break.

Bearer tokens with redirect work and will continue to work fine.  It likely means that your login will have to be done in a popup or an IFRAME, as well as back channel communication (identity provider to service provider directly) if you want to keep a single page application.

SAML isn't broken.

**Comment by derhelge** [comment]
Hi @xet7,

any news on this? we paid on implementing SAML/Shibboleth-Authentication.

**Comment by xet7** [comment]
@derhelge 

I need some more details:

1. With SAML/Shibboleth-Authentication, do you mean they are some same thing, or different?
2. Do you mean SAML similar like in RocketChat ?
3. Or do you mean SAML similar like in https://sandstorm.io ?
4. Or do you mean SAML/Shibboleth-Authentication similar like in some other software?
5. If you mean Shibboleth-Authentication, is there example NPM or https://atmospherejs.com package that is compatible with your Identity Provider?
6. What Identity Provider you use, so that I can test compatibility?

**Comment by xet7** [comment]
7. In SAML/Shibboleth-Authentication, is there need to map some Organizations/Teams/People stucture to corresponding WeKan Organizations/Teams/People structure? Can you send example anonymized example query result about structure to support@wekan.team ?
